vmm/src/vm.rs

```diff
@@ -511,7 +511,9 @@ impl Vm {
         let sev_snp_enabled = config.lock().unwrap().is_sev_snp_enabled();
         #[cfg(feature = "tdx")]
         let force_iommu = tdx_enabled;
-        #[cfg(not(feature = "tdx"))]
+        #[cfg(feature = "sev_snp")]
```

Comment:
I feel like this isn't quite right. My premise here is TDX and SEV-SNP are not mutually exclusive.

If both are enabled, when TDX is enabled while SEV-SNP is not, the valuable is going to be set to false, which breaks TDX.

The `cfg` tests should be more complex.

1. If both TDX and SEV-SNP are enabled in the build, `force_iommu` should be set to `tdx_enabled || sev_snp_enabled`.
2. If either is enabled, you only use their respective variable.
3. If neither is enabled, you set `force_iommu` to false.

Reply:
We have a compile time check to make sure that TDX and SEV-SNP are mutually exclusive. https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/src/main.rs#L759

Reply:
Okay, there is a safeguard so this code is fine.